### ISSUE TITLE
Improving OSPR Management

### ISSUE THREAD

**Issue by cassiezamparini** [openedx/wg-coordination#134]
See proposal: https://openedx.atlassian.net/wiki/spaces/COMM/pages/4516577281/Improving+OSPR+Management

**Comment by antoviaque** [comment]
@xitij2000 @itsjeyd During the last meeting of the coordination working group, we set as a goal to have a plan for each of the approved proposals from the summit. It's likely going to be tricky to get things done with Christmas, but what would be your plan & timeline beyond that?

**Comment by itsjeyd** [comment]
@antoviaque This sprint I spent some time reviewing the scope of the proposal and scheduling tickets for implementing it. There are three different buckets of work, if you will:

1. Documentation updates
    * Tickets ready to be worked on
    * Tentative completion date: Feb 28 2025
1. Implementation of features and processes that are unique to this proposal
    * Tickets ready to be worked on
    * Tentative completion date: Apr 30 2025
1. Implementation of features and processes that overlap with #135 and #140 [fully](https://openedx.atlassian.net/wiki/spaces/COMM/pages/4516577281/Improving+OSPR+Management#Automating-assignment-of-OSPRs) or [partially](https://openedx.atlassian.net/wiki/spaces/COMM/pages/4516577281/Improving+OSPR+Management#Making-CC-info-for-individual-repositories-easily-discoverable).
    * No tickets yet
    * Next step will be to have a conversation with @bradenmacdonald and @navinkarkera to see how we can combine the suggested approaches from the three proposals.

CC @xitij2000 @cassiezamparini

**Comment by itsjeyd** [comment]
FYI @mphilbrick211 @antoviaque 

### Status update

Here is a comprehensive list of tickets we have scheduled for the new year:

#### Documentation updates

* Update documentation about PRs needing product review ([details](https://gist.github.com/itsjeyd/90ad745f8131f68c01a3d0aca3f93b60))
* Improve documentation on how to find OSPRs to review ([details](https://gist.github.com/itsjeyd/a4b5b2f987e2199ed01931599e57449f))
* Simplify the OSPR welcome message ([details](https://gist.github.com/itsjeyd/902416166f1509ea9897c1f3d95faf5e))

#### Implementation of features and processes that are unique to this proposal

* Create prototype app to list all core contributors for a repo ([details](https://gist.github.com/itsjeyd/8f1089b5f394cb0fdb1fde4e255a8543))
* Discovery: How to get a global list of CCs to use for automation purposes ([details](https://gist.github.com/itsjeyd/4afb5c80fadba32b7d32df5ee474972b))
* Automatically remove obsolete labels from OSPRs upon merge/close ([details](https://gist.github.com/itsjeyd/bae67534105cdfdffb2b3dcf959c58a9))
* Automatically populate metadata on Contributions board ([details](https://gist.github.com/itsjeyd/63efe2c7e64eb6d3e7a1f92967238904))

#### Implementation of features and processes that overlap with other proposals

* Define workflow for interactive PR bot that enacts Single Responsible Person Principle ([details](https://gist.github.com/itsjeyd/099d0bb6101b7ee16c110434113178fa))

---

@cassiezamparini I would love to turn those into actual GitHub tickets but wasn't sure where to put them?

---

CC @xitij2000 @bradenmacdonald @navinkarkera

**Comment by cassiezamparini** [comment]
> I would love to turn those into actual GitHub tickets but wasn't sure where to put them?

@itsjeyd Hmmm, a good question. I'm going to defer this to @antoviaque and @jalondonot. What is best practice here? I don't want the issues to get confused with our WG issues, so perhaps this needs to be a separate project?

**Comment by antoviaque** [comment]
@cassiezamparini What would be the issue with having the tickets as part of the working group? They are effectively subtasks or split tasks from one already on the board, no?

**Comment by cassiezamparini** [comment]
@antoviaque I just thought the board could get very cluttered: https://github.com/orgs/openedx/projects/62/views/1. But I guess it would be a good way to keep tabs on the projects as we could sift through the tickets in our meetings...

**Comment by antoviaque** [comment]
@cassiezamparini True, that could happen if we overdid it - in which case we might be able to address it with different views of the board, depending on the granularity we want during the meeting?

**Comment by cassiezamparini** [comment]
@antoviaque Makes sense.

FYI. @itsjeyd You can create issues on the Contributors Coordination Topics board.

**Comment by itsjeyd** [comment]
@cassiezamparini Sounds good, I'll do that. Thanks to you and @antoviaque for clarifying 👍

**Comment by itsjeyd** [comment]
Tickets are up on the board now:

* https://github.com/openedx/wg-coordination/issues/145
* https://github.com/openedx/wg-coordination/issues/146
* https://github.com/openedx/wg-coordination/issues/147
* https://github.com/openedx/wg-coordination/issues/148
* https://github.com/openedx/wg-coordination/issues/149
* https://github.com/openedx/wg-coordination/issues/150

**Comment by itsjeyd** [comment]
Tickets are up on the board now:

* https://github.com/openedx/wg-coordination/issues/145
* https://github.com/openedx/wg-coordination/issues/146
* https://github.com/openedx/wg-coordination/issues/147
* https://github.com/openedx/wg-coordination/issues/148
* https://github.com/openedx/wg-coordination/issues/149
* https://github.com/openedx/wg-coordination/issues/150
* https://github.com/openedx/wg-coordination/issues/151
* https://github.com/openedx/wg-coordination/issues/152

**Comment by antoviaque** [comment]
@cassiezamparini @itsjeyd 

> I just thought the board could get very cluttered: https://github.com/orgs/openedx/projects/62/views/1

Btw, to add explicit sub-tasks without cluttering, maybe github officially supporting those will help?

https://github.blog/changelog/2025-01-13-evolving-github-issues-public-preview/#break-down-and-nest-issues-with-sub-issues

**Comment by itsjeyd** [comment]
@antoviaque Oh, awesome! Such a timely move from GitHub to be adding this functionality now 😎 

I'll try it out.

CC @cassiezamparini

**Comment by itsjeyd** [comment]
@antoviaque @cassiezamparini It looks like the sub-issues still show up as standalone tickets on the board. But either way, having a list of all the issues that belong to this proposal attached to the first post is probably still going to be helpful.

**Comment by antoviaque** [comment]
@itsjeyd It's probably a good thing they _can_ show up on the board - maybe they can be filtered out from specific board though? I haven't looked in detail, so I am just speculating.

**Comment by antoviaque** [comment]
-snip- -- I was wrong.

**Comment by itsjeyd** [comment]
@antoviaque Your hunch was right, there is a way to enable filtering, by setting the `Type` field of each sub-issue to `Task`. Then you can say `-type:Task` on the [board](https://github.com/orgs/openedx/projects/62/views/1?filterQuery=-type%3ATask) and it will hide the sub-issues.

![task-dialog](https://github.com/user-attachments/assets/a2088baf-357c-4eba-affb-abbc873d8544)

CC @cassiezamparini

**Comment by itsjeyd** [comment]
@cassiezamparini Can I ask you to please add this filter to the ticket description:

[Tasks from this epic](https://github.com/orgs/openedx/projects/62/views/1?filterQuery=type:Task+parent-issue:%22openedx/wg-coordination%23134%22)

**Comment by cassiezamparini** [comment]
Done @itsjeyd 😄

**Comment by itsjeyd** [comment]
Great, thanks @cassiezamparini 🚀

**Comment by antoviaque** [comment]
@itsjeyd For the contributors meeting tomorrow, would you have an update on the overall progress of this work? I notice that several tasks have been closed, so it looks like it is making good progress, but an overview/summary would be useful :)

**Comment by itsjeyd** [comment]
@antoviaque Sure!

We're almost done implementing this proposal.

* The OSPR welcome message has been reorganized to better highlight the most important information, while retaining more detailed info in sections that are collapsed by default.
* Finding criteria for determining if an OSPR needs product review is now straightforward.
* Axim took care of making CC info for individual repositories more easily discoverable by introducing repo-specific CC teams. (To find the list of CCs for a repo, just go to https://github.com/orgs/openedx/teams and search for `committers-<repo-name>`.)
* Most manual OSPR management tasks listed in the proposal have been automated. (As a result of these automations, the time that it takes me to do a full triage pass over OSPRs that I'm a PM for has  gone down quite a bit.)
    * Removing obsolete labels when an OSPR gets merged/closed.
    * Populating `Date opened` field on Contributions board.
    * Populating `Repo Owner / Owning Team` field on Contributions board.
    * Populating new `Date merged / closed` field on Contributions board.
 * We have a plan for automating the process of managing OSPRs against maintained repos. Coming up with this plan was shared work between this epic and #135. Implementation of the plan will happen via #135.
   
In terms of work left to do for this epic, we have:

* (Documentation update) #146 
* (Automation task) #155 

The latter might be blocked waiting for budget for a little while.

CC @xitij2000 @bradenmacdonald @navinkarkera
